client/src/core/model-registry.js

```diff
@@ -311,90 +311,104 @@ class ModelRegistry {
       color = {r: 255, g: 255, b: 255},
       bcolor = {r: 0, g: 0, b: 255}) {
     const canvas = document.createElement('canvas');
-    canvas.width = ratio > 1 ? 256 : 256 * ratio;
-    canvas.height = ratio > 1 ? 256 / ratio : 256;
+    const canvasWidth = ratio > 1 ? 256 : 256 * ratio;
+    const canvasHeight = ratio > 1 ? 256 / ratio : 256;
```

Comment:
Maybe store this `256` into a dedicated const variables at the top of the file, like `maxCanvasWidth` and `maxCanvasHeight` or something like this